tests/widget/modal/widget-modal.login-social-redirect.test.js

```diff
@@ -0,0 +1,15 @@
+import { expect, test } from '@playwright/test';
+
+test.only('Modal widget with social callback redirect', async ({ page }) => {
+  await page.goto('widget/modal?journey=LoginWithSocial');
+
+  const loginButton = page.locator('button', { hasText: 'Open Login Modal' });
+  await loginButton.click();
+
+  await page.fill('text="User Name"', 'demouser');
```

Comment:
This is probably not needed for the test-case of using Google, but it brings up a good point. We have logic in the framework that manages self-submission and user-input optionality. We should test these permutations. 

So, having a test that locally logs a user in, with the presence of a social login option would be good. It would be good to test the absence of the local auth button. Then, we should have a test for local auth, but not form, which would show a local auth button.

Reply:
Yes I noticed this and thought that was a mistake. However yes I think you're right, testing the combinations of social auth, local auth, social auth without any local auth etc makes sense. I'll build this out.